### PR TITLE
Support Sles 15 Sp3, RHEL 8, Ubunutu 18.04 and 20.04

### PR DESCRIPTION
**Pull request by mashoodk**
Please note that with this commit other operating systems will not be
     supported.

### CODE REVIEW

File: src/pmi/runtime_pmix.h [hunk 1]
```diff
@@ -61,7 +61,8 @@ class Pmix_Runtime : public Fam_Runtime {
             return rc;
         }
         mInitrc = rc;
-        (void)strncpy(proc.nspace, mProc.nspace, PMIX_MAX_NSLEN);
+        //(void)strncpy(proc.nspace, mProc.nspace, PMIX_MAX_NSLEN);
```
Comment (BA-Ramya) [inline src/pmi/runtime_pmix.h:64]: We can take off this comment at line 64

File: test/common/fam_test_config.h.in [hunk 2]
```diff
@@ -51,7 +51,7 @@ using namespace openfam;
 #cmakedefine TEST_FAM_CONTEXT_MODEL "${TEST_FAM_CONTEXT_MODEL}"
 #cmakedefine TEST_RUNTIME "${TEST_RUNTIME}"
 #cmakedefine TEST_NUM_CONSUMER "${TEST_NUM_CONSUMER}"
-
+#cmakedefine USE_CONFIG_OPTIONS "${USE_CONFIG_OPTIONS}"
 #ifdef MEMSERVER_PROFILE
 #cmakedefine TEST_MEMSERVER_IP "0:127.0.0.1"
 #endif
```
Comment (BA-Ramya) [inline test/common/fam_test_config.h.in:57]: Does line 55-57 (TEST_MEMSERVER_IP) also require change if USE_CONFIG_OPTIONS is set?

File: third-party/build.sh [hunk 3]
```diff
@@ -65,67 +97,87 @@ fi
...
 
 esac		
-
-pip3 install ruamel.yaml
+if [ "$no_package_install" != "true" ] 
+then
+    pip3 install ruamel.yaml
+fi
 
 echo "Finished installing required RPMS"
```
Comment (BA-Ramya) [inline third-party/build.sh:179]: Message is required only if  [ "$no_package_install" != "true" ] . So push it within the if conditional after pip3 install

File: third-party/build.sh [hunk 4]
```diff
@@ -135,23 +187,28 @@ cd grpc
 git submodule update --init
 
 #For parallel compilation, make -j is selected. For systems with limited DRAM, avoid using -j option for all make commands
-make -j 
+#make -j 
+mkdir -p cmake/build
+cd cmake/build
+cmake ../.. -DBUILD_SHARED_LIBS=ON -DgRPC_INSTALL=ON -DCMAKE_INSTALL_PREFIX=$ABS_BUILD_DIR
+#cmake ../.. -DgRPC_INSTALL=ON -DCMAKE_INSTALL_PREFIX=$ABS_BUILD_DIR
+make -j
```
Comment (BA-Ramya) [inline third-party/build.sh:195]: make -j will fail on a system with lower memory. And the errors appear as internal compiler error and do not give enough hints to analyze build failure. This problem is often seen with nvmm build.
Should we add a check to obtain memory size? If DRAM size is greater than some threshold(say 32GB), then issue a make -j. Else use make.

File: third-party/download.sh [hunk 5]
```diff
@@ -69,10 +69,11 @@ rm -fr googletest_v1_10_x
 #NVMM
 cd $CURRENT_DIR
 echo "Downloading NVMM source"
-git clone https://github.com/HewlettPackard/gull.git nvmm
+#git clone https://github.com/HewlettPackard/gull.git nvmm
+git clone -b devel https://github.com/mashoodk/gull.git nvmm
```
Comment (BA-Ramya) [inline third-party/download.sh:73]: This need to be replaced with public repo

File: third-party/download.sh [hunk 6]
```diff
@@ -69,10 +69,11 @@ rm -fr googletest_v1_10_x
...
 cd nvmm
-git fetch --all --tags --prune
-git checkout tags/v0.1 -b openfam
+#git fetch --all --tags --prune
+#git checkout tags/v0.1 -b openfam
```
Comment (BA-Ramya) [inline third-party/download.sh:76]: Don't we need these any more?

File: third-party/nvmm_build.sh [hunk 7]
```diff
@@ -56,22 +56,23 @@ case $OS in
...
-        exit 1
-fi
-make install prefix="$CWD/../build"
-if [[ $? > 0 ]]
-then
-        echo "nvml make install failed, exiting..."
-        exit 1
-fi
-cd ../../build
+#cd third-party/
```
Comment (BA-Ramya) [inline third-party/nvmm_build.sh:59]: Please check if nvml build instructions needs to be uncommented

**Review by faizan-barmawer — COMMENTED** [review]
Few minor comments.
Rest all changes look fine.

Regards,
Faizan

**Review by faizan-barmawer — APPROVED** [review]
Code changes look good to merge.

Regards,
Faizan

**Pull request merged**